File: domain/src/main/java/mixter/domain/core/message/handlers/UpdateTimeline.java
package mixter.domain.core.message.handlers;

import mixter.doc.Handler;
import mixter.domain.core.message.TimelineMessageProjection;
import mixter.domain.core.message.TimelineMessageRepository;
import mixter.domain.core.message.events.MessageDeleted;
import mixter.domain.core.message.events.MessageQuacked;

@Handler
public class UpdateTimeline {
    private TimelineMessageRepository timelineRepository;

    public UpdateTimeline(TimelineMessageRepository timelineRepository) {

        this.timelineRepository = timelineRepository;
    }

    public void apply(MessageQuacked event) {
        timelineRepository.save(new TimelineMessageProjection(event.getAuthorId(), event.getAuthorId(), event.getMessage(), event.getMessageId()));
    }

    public void apply(MessageDeleted messageDeleted) {
        timelineRepository.delete(messageDeleted.getId());
    }
}

File: domain/src/main/scala/mixter/domain/message/handlers/UpdateTimeline.scala
package mixter.domain.message.handlers

import mixter.domain.message.event.{MessageDeleted, MessageQuacked}
import mixter.domain.message.{TimelineMessageProjection, TimelineMessageRepository}

class UpdateTimeline(repository: TimelineMessageRepository) {
  def apply(event: MessageQuacked): Unit = {
    repository.save(TimelineMessageProjection(event.author, event.author, event.message, event.id))
  }
  def apply(event: MessageDeleted): Unit = {
    repository.delete(event.id)
  }
}

